README.md

```diff
@@ -1,87 +1,121 @@
+<label>Post Description</label><br>
+<%= f.text_area :description %><br>
+```
 
-form_for(@post) do |form_builder| <%= form_builder.text_field :title %> <%= form_builder.text_area :content %> <%= form_builder.submit %> end
+Isn't that much cleaner? Notice how we no longer have to pass in the values manually? By passing in the attribute as a symbol (e.g. `:title`) that will automatically tell the form field what model attribute to bind to, it also is what auto fills the values for us. Now let's refactor the submit button, instead of `<%= submit_tag "Submit Post" %>` we can change it to:
 
-submit the form and we don't need to update the controller action because of mass-assignment.
+```ERB
+<%= f.submit %>
+```
 
-That in a nutshell is how form_for works.
+Lastly, `form_for` also automatically sets the `authenticity_token` value for us, so we can remove the `<%= hidden_field_tag :authenticity_token, form_authenticity_token %>` line completely. Our new form will look something like this:
 
-other form builders, select_tag, hidden_tag, etc.
-options to pass to form_for, most important is URL
+```ERB
+<h3>Post Form</h3>
+
+<%= form_for(@post) do |f| %>
+  <label>Post title:</label><br>
+  <%= f.text_field :title %><br>
+
+  <label>Post Description</label><br>
+  <%= f.text_area :description %><br>
+  
+  <%= f.submit %>
+<% end %>
+```
+
+Our refactor work isn't quite done, if you had previously created a `PUT` route like we did in the `form_tag` lesson, we'll need to change that to a `PATCH` method since that is the HTTP verb that `form_for` utilizes, we can make that change in the `config/routes.rb` file:
+
+```ruby
+patch 'post/:id', to: 'posts#update'
+```
+
+Now if you start the Rails server and go to an edit page you'll see that the data is loaded into the form and everything appears to be working properly, however if you change the value of one of the form fields and click `Update post` you will see that the record isn't updated. So what's happening? When I run into behavior like this I'll usually look at the console logs to see if it tells me anything. Below is the screenshot of what I see after submitting the form:
+
+![Unpermitted Parameters](https://s3.amazonaws.com/flatiron-bucket/readme-lessons/unpermitted_params.png)
+
+If you remember back to the original [edit lesson](https://github.com/learn-co-curriculum/rails-edit-update-action-readme) you will remember that Rails 4 requires us to secure our forms by integrating strong parameters. If you look up at the console log you'll see that one of the parameters that is unpermitted is `post`. This is because `form_for` is bound directly with the `Post` model, so we need to pass the model into the strong parameter call in the controller. So let's change: `@post.update(params.permit(:title, :description))` to:
```

Comment:
Also isn't this change because of how the params get passed in? With our old form it was just

```
{
"title": "blah",
"description": "blah blah"
}
```

and with form for it's

````
{
"post": {
            "title": "blah",
            "description": "blah blah"
            }
}
```

I could be wrong, but I think that's a bit more a of visceral reason why you need the `require`